libvast/vast/logger.hpp

```diff
@@ -44,22 +44,28 @@
 #include "vast/detail/logger.hpp"
 #include "vast/detail/logger_formatters.hpp"
 
-#define VAST_DEBUG(...) SPDLOG_LOGGER_TRACE(vast::detail::logger(), __VA_ARGS__)
+#define VAST_DEBUG(...)                                                        \
+  SPDLOG_LOGGER_TRACE(::vast::detail::logger(), __VA_ARGS__)
 #define VAST_VERBOSE(...)                                                      \
-  SPDLOG_LOGGER_DEBUG(vast::detail::logger(), __VA_ARGS__)
-#define VAST_INFO(...) SPDLOG_LOGGER_INFO(vast::detail::logger(), __VA_ARGS__)
-#define VAST_WARN(...) SPDLOG_LOGGER_WARN(vast::detail::logger(), __VA_ARGS__)
-#define VAST_ERROR(...) SPDLOG_LOGGER_ERROR(vast::detail::logger(), __VA_ARGS__)
+  SPDLOG_LOGGER_DEBUG(::vast::detail::logger(), __VA_ARGS__)
+#define VAST_INFO(...) SPDLOG_LOGGER_INFO(::vast::detail::logger(), __VA_ARGS__)
+#define VAST_WARN(...) SPDLOG_LOGGER_WARN(::vast::detail::logger(), __VA_ARGS__)
+#define VAST_ERROR(...)                                                        \
+  SPDLOG_LOGGER_ERROR(::vast::detail::logger(), __VA_ARGS__)
 #define VAST_CRITICAL(...)                                                     \
-  SPDLOG_LOGGER_CRITICAL(vast::detail::logger(), __VA_ARGS__)
+  SPDLOG_LOGGER_CRITICAL(::vast::detail::logger(), __VA_ARGS__)
 
 #if VAST_LOG_LEVEL >= VAST_LOG_LEVEL_TRACE
 
 #  define VAST_TRACE(...)                                                      \
-    VAST_DEBUG(vast::detail::spd_msg_from_args(1, 2, __VA_ARGS__).str(),       \
-               "ENTER", __func__, __VA_ARGS__);                                \
-    auto CAF_UNIFYN(vast_log_trace_guard_) = ::caf::detail::make_scope_guard(  \
-      [=, func_name_ = __func__] { VAST_DEBUG("EXIT {}", func_name_); })
+    auto CAF_UNIFYN(vast_log_trace_guard_)                                     \
+      = [func_name_ = __func__](auto&& format, auto&&... args) {               \
+          VAST_DEBUG(::fmt::format("ENTER {} {}", func_name_,                  \
+                                   std::forward<decltype(format)>(format)),    \
+                     std::forward<decltype(args)>(args)...);                   \
```

Comment:
Would this work for an invocation like `VAST_TRACE(1, 2)`, i.e. without format string? Or was this previously unsupported as well?

Reply:
That isn't supported by any of the other logging macros either. It was previously supported with the old CAF logger, but that didn't require format strings.

Reply:
This never worked like any of the other logging macros anyways, so I wouldn't expect symmetry.

I'm asking because if we require a format string anyways, it seems like this could just be written as 
```
[ func_name_ = __func__](std::string_view format, auto&&... args) {
  VAST_DEBUG("ENTER {} " + format, func_name_, std::forward<decltype(args)>(args)...);
}
```

Reply:
Looks like it should work, let's see.